app/gui2/src/components/GraphEditor/GraphVisualization.vue

```diff
@@ -36,6 +39,62 @@ const emit = defineEmits<{
   setVisualizationVisible: [visible: boolean]
 }>()
 
+const visPreprocessor = ref(DEFAULT_VISUALIZATION_CONFIGURATION)
+const error = ref<Error>()
+
+const projectStore = useProjectStore()
+const graphStore = useGraphStore()
+const visualizationStore = useVisualizationStore()
+
+const expressionInfo = computed(
+  () => props.expressionId && graphStore.db.getExpressionInfo(props.expressionId),
+)
+const typeName = computed(() => expressionInfo.value?.typename ?? 'Any')
+
+/** The Visualization ID for the method to get the default visualization. */
+const defaultVisualizationVisualizationId = random.uuidv4() as Uuid
+const defaultVisualization = computedAsync(async () => {
+  if (props.currentType) return
+  if (!props.expressionId) return
+  const ident = props.pattern?.repr()
+  if (!ident) return
+  const ls = await projectStore.lsRpcConnection
+  const dataServer = await projectStore.dataConnection
+  ls.executeExpression(
+    projectStore.executionContext.id,
+    defaultVisualizationVisualizationId,
+    props.nodeId as ExprId,
+    `${ident}.default_visualization.to_js_object.to_json`,
+  )
```

Comment:
Node's pattern may be many things. If it's `Point x y`, the expression would be `Point x y..default_visualization.to_js_object.to_json` what may give quite unexpected result.

And I think we should attach a "normal" visualization - this way we will just get the node's value without need of retrieving the node's pattern. And we could then react to value changes, for example when the Table stops having `longitude` and `latitude` columns, the default vis could change to `Table`.

Reply:
that makes sense. actually i tried using a normal visualization before, but i'm not sure how to do dynamic dispatch (`.default_visualization.to_js_object.to_json`) on a visualization. i'll get back to you when i have some errors to show

Reply:
I think the preprocessor should be just `a -> a.default_visualization.to_js_object.to_json`

If this does not work, then we need a new method in `Visualizations` module in enso stdlib.

Reply:
hmm... i wonder if `(${ident}).default_visualization.to_js_object.to_json` would be sufficient. but attaching a normal visualization is definitely better